src/main/scala/firrtl/backends/verilog/VerilogEmitter.scala

```diff
@@ -467,6 +469,9 @@ class VerilogEmitter extends SeqTransform with Emitter {
     def getMemoryEmissionOption(target: ReferenceTarget): MemoryEmissionOption =
       memoryEmissionOption(target)
 
+    def getMemoryInitEmissionOption(): MemoryInitEmissionOption =
+      MemoryInitBlockEmissionOption
+
```

Comment:
```suggestion
    def emitMemoryInitAsNoSynth: Boolean =
      annotations.collectFirst { case SomeAnnotationIndicatingWhatToDo => true }.getOrElse(false)
      MemoryInitBlockEmissionOption
      
```
I think you can then just add a `case object SomeAnnotationIndicatingWhatToDo extends NoTargetAnnotation` (pick a better name) and get rid of the `MemoryInitEmissionValue` and `MemoryInitEmissionOption` stuff above. This would make it pretty easy to control whether the emission style.

Minor style nit but `()` at the end of a 0-arity method implies it mutates something which this does not.

Reply:
I also thought it became a little convoluted (just a little? 😂).
Gonna try to simplify it.

Reply:
This relates to the comments above on `private val memoryInitEmissionOption`, I would make this:
```scala
// Somewhere else
case object EmitMemoryInitBlockOutsideIfDefAnnotation extends NoTargetAnnotation
...

// Now in this code block
lazy val emitMemoryInitAsNoSynth: Boolean = annotations.collectFirst {
  case EmitMemoryInitBlockOutsideIfDefAnnotation => true
}.getOrElse(false)
// Then you can just use this value elsewhere in the emitter
```

@ekiwi thoughts on the annotation name / style? It could be a `case class` with a Boolean argument or as I suggested, default to inside the ifndef SYNTHESIS and emit outside of it if the annotation is provided

Reply:
We can do a single case object. Maybe something like `EmitMemoryInitForSynthesis` or `SynthesizeMemoryInit`.

Reply:
I added both cases so one could define in the code which anno to use.